core/src/dpdk/port.rs

```diff
@@ -437,26 +474,57 @@ impl<'a> PortBuilder<'a> {
             }
 
             // configures the TX queue with defaults
-            let txq_index = TxQueueIndex(idx as u16);
+            let txq = TxQueueIndex(idx as u16);
             unsafe {
                 ffi::rte_eth_tx_queue_setup(
                     self.port_id.0,
-                    txq_index.0,
+                    txq.0,
                     self.txd,
                     socket_id.0 as raw::c_uint,
                     ptr::null(),
                 )
                 .to_result()?;
             }
 
-            let queue = PortQueue {
-                port_id: self.port_id,
-                rxq_index,
-                txq_index,
-                kni: kni.as_ref().map(|v| v.txq()),
-            };
+            let mut q = PortQueue::new(self.port_id, rxq, txq);
+
+            if let Some(kni) = &kni {
+                q.set_kni(kni.txq());
+            }
+
+            #[cfg(feature = "metrics")]
+            {
+                // have space to set up the stats per core.
+                if ffi::RTE_ETHDEV_QUEUE_STAT_CNTRS >= len as u32 {
```

Comment:
minor: maybe a line to this dpdk call in their docs?

Reply:
assumption throughout the codebase is for any ffi calls into dpdk, to get more you should just go look up the dpdk docs. maybe we will add that tidbit to the README for the ffi crate later. trying to avoid making the already fairly messy codebase messier.